### PR TITLE
Add `chainerx.fmod`

### PR DESCRIPTION
**Pull request by aksub99**
Add `chainerx.fmod`

### CODE REVIEW

**Comment by asi1024** [comment]
#7912

**Comment by take-cheeze** [comment]
Jenkins, test this please!

**Comment by take-cheeze** [comment]
@aksub99 
BTW did you see https://github.com/chainer/chainer/issues/7914 ?

**Comment by aksub99** [comment]
Yes @take-cheeze . I did take a look at #7914 . I relaxed the backward tolerance from 5e-2 to 5e-3. The tests are not flaky in my environment now. Should I make any more modifications?

File: tests/chainerx_tests/unit_tests/routines_tests/test_arithmetic.py [hunk 1]
```diff
@@ -1404,3 +1405,69 @@ class TestReciprocal(math_utils.UnaryMathTestBase, op_utils.NumpyOpTest):
...
+            {'atol': 1e-3, 'rtol': 1e-2})
+
+    def generate_inputs(self):
+        shape1, shape2 = self.in_shapes
+        dtype1, dtype2 = self.in_dtypes
+        a, b = super().generate_inputs()
+        # division with too small divisor is unstable.
+        for i in numpy.ndindex(shape1):
+            if math.fabs(b[i]) < 0.1:
+                b[i] += 1.0
```
Comment (asi1024) [inline tests/chainerx_tests/unit_tests/routines_tests/test_arithmetic.py]: How about writing as follows for code simplification?
```py
b[numpy.abs(b) < 0.1] += 1.0
```

File: tests/chainerx_tests/unit_tests/routines_tests/test_arithmetic.py [hunk 2]
```diff
@@ -1404,3 +1405,69 @@ class TestReciprocal(math_utils.UnaryMathTestBase, op_utils.NumpyOpTest):
 
     def func(self, xp, a):
         return xp.reciprocal(a)
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'in_shapes': [((2, 3), (2, 3))],
```
Comment (asi1024) [inline tests/chainerx_tests/unit_tests/routines_tests/test_arithmetic.py]: Please add tests for broadcast.

File: tests/chainerx_tests/unit_tests/routines_tests/test_arithmetic.py [hunk 3]
```diff
@@ -1404,3 +1405,69 @@ class TestReciprocal(math_utils.UnaryMathTestBase, op_utils.NumpyOpTest):
...
+        'in_dtypes,out_dtype': (
+            dtype_utils.make_same_in_out_dtypes(
+                2, chainerx.testing.numeric_dtypes)),
+        'input_lhs': ['random'],
```
Comment (take-cheeze) [inline tests/chainerx_tests/unit_tests/routines_tests/test_arithmetic.py]: Can you add tests for negative values?

File: tests/chainerx_tests/unit_tests/routines_tests/test_arithmetic.py [hunk 4]
```diff
@@ -1404,3 +1405,69 @@ class TestReciprocal(math_utils.UnaryMathTestBase, op_utils.NumpyOpTest):
...
+        'input_rhs': ['random'],
+        'is_module': [True, False],
+    })
+    # Special values
+    + chainer.testing.product({
+        'in_shapes': [((2, 3), (2, 3))],
+        'in_dtypes,out_dtype': (
+            dtype_utils.make_same_in_out_dtypes(
+                2, chainerx.testing.float_dtypes)),
+        'input_lhs': ['random', float('inf'), -float('inf'), float('nan')],
```
Comment (take-cheeze) [inline tests/chainerx_tests/unit_tests/routines_tests/test_arithmetic.py]: Please add non-differential test case

File: chainerx_cc/chainerx/native/native_device/arithmetic.cc [hunk 5]
```diff
@@ -235,6 +235,8 @@ class NativePowerSAKernel : public PowerSAKernel {
 
 CHAINERX_NATIVE_REGISTER_KERNEL(PowerSAKernel, NativePowerSAKernel);
 
+CHAINERX_NATIVE_REGISTER_ELTWISE_BINARY_KERNEL(FmodKernel, { out = chainerx::Fmod(x1, x2); });
```
Comment (take-cheeze) [inline chainerx_cc/chainerx/native/native_device/arithmetic.cc]: Please add overloading to integer types like remainder:
https://github.com/chainer/chainer/pull/7675/files#diff-cc98d139a41fe00c17b77a83d1ae42dfR238

**Comment by asi1024** [comment]
I'll takeover this PR: #8110
